platform/ecs/agent/client.go

```diff
@@ -90,6 +91,11 @@ func (c *client) newRequest(endpoint string) (*http.Request, error) {
 }
 
 func decodeBody(resp *http.Response, out interface{}) error {
-	defer resp.Body.Close()
+	if resp.Body != nil {
```

Comment:
It's unnecessary to check resp.Body is not nil after checking the error of client.Do. If resp.Body is nil, `json.NewDecoder(resp.Body).Decode(out)` panics.

Reply:
Thanks for explaining it, now I understand!